code/go/0chain.net/smartcontract/errors.go

```diff
@@ -0,0 +1,136 @@
+package smartcontract
```

Comment:
Even though the issue title suggested response status specific to smart contracts, this way of API error reporting looks good and is something we could share across different parts of our application. It would make our error reporting more consistent. Just a thought. Maybe we can do that later?

Reply:
Sure, I've moved the main errors, corresponds to http statuses, to the `common` package.

Reply:
@MurashovVen 
Maybe lets try avoiding `common` and `util` package nameing, as at some point those packages become harder to identify their purpose. Better have domain specific error separation.
https://blog.golang.org/package-names

> Avoid meaningless package names. Packages named util, common, or misc provide clients with no sense of what the package contains. This makes it harder for clients to use the package and makes it harder for maintainers to keep the package focused. Over time, they accumulate dependencies that can make compilation significantly and unnecessarily slower, especially in large programs. And since such package names are generic, they are more likely to collide with other packages imported by client code, forcing clients to invent names to distinguish them.